spec/system/user_management_spec.rb

```diff
@@ -3,77 +3,70 @@
 require 'rails_helper'
 
 RSpec.describe 'User Management' do
+  let!(:user) { create(:user) }
+
   before do
-    @user = create(:user)
     when_current_user_is :admin
+    visit users_path
   end
 
-  context 'deleting a user unsuccessfully', :js do
-    before do
-      @log = create(:log_entry, user: @user)
+  context 'when deleting a user', :js do
+    subject :submit do
+      page.accept_confirm 'Are you sure?' do
+        within('tr', text: user.name) { click_on 'Destroy' }
+      end
     end
 
-    it 'puts the errors in the flash' do
-      visit users_path
-      # Click the 'destroy' link in the user's row
-      page.accept_confirm 'Are you sure?' do
-        within('tr', text: @user.name) { click_on 'Destroy' }
+    context 'when a user is not deletable' do
+      before { create(:log_entry, user:) }
+
+      it 'puts the errors in the flash' do
+        submit
+        expect(page).to have_text('Cannot delete record because dependent log entries exist')
       end
-      expect(page).to have_text(
-        'Cannot delete record because dependent log entries exist'
-      )
     end
-  end
 
-  context 'deleting a user successfully', :js do
-    it 'deletes the user and says it did' do
-      visit users_path
-      page.accept_confirm 'Are you sure?' do
-        within('tr', text: @user.name) { click_on 'Destroy' }
+    context 'when a user is deletable' do
+      it 'deletes the user and says it did' do
+        submit
+        expect(page).to have_text('User successfully destroyed')
       end
-      expect(page).to have_text(
-        'User successfully destroyed'
-      )
     end
   end
 
-  context 'editing an existing user successfully' do
+  context 'when editing a user' do
+    before do
+      within('tr', text: user.name) { click_on 'Edit' }
+    end
+
     it 'updates the user' do
-      visit users_path
-      within('tr', text: @user.name) { click_on 'Edit' }
       fill_in 'Name', with: 'Bar Foo'
-      click_button 'Save'
-      expect(page).to have_text 'User successfully updated'
+      click_on 'Save'
+      expect(page).to have_text('User successfully updated')
     end
-  end
 
-  context 'editing an existing user unsuccessfully' do
     it 'puts the error in the flash' do
-      visit users_path
-      within('tr', text: @user.name) { click_on 'Edit' }
       fill_in 'Spire', with: 'not a valid spire'
-      click_button 'Save'
-      expect(page).to have_text 'Spire must be 8 digits followed by @umass.edu'
+      click_on 'Save'
+      expect(page).to have_text('Spire must be 8 digits followed by @umass.edu')
     end
```

Comment:
Did you drop the successful/unsuccessful contexts on purpose?

Reply:
Yeah, I think context descriptions like these don't make sense. They don't describe "context", they describe the examples they contain. They were also empty except for the one example they each contained.

Reply:
Cool, just wanted to check because seeing set up stuff inside an example sets off bells in my brain.